server/conn.go

```diff
@@ -2090,7 +2090,20 @@ func (cc *clientConn) handleStmt(ctx context.Context, stmt ast.StmtNode, warns [
 	ctx = context.WithValue(ctx, util.ExecDetailsKey, &util.ExecDetails{})
 	reg := trace.StartRegion(ctx, "ExecuteStmt")
 	cc.audit(plugin.Starting)
-	rs, err := cc.ctx.ExecuteStmt(ctx, stmt)
+	var rs ResultSet
+	var err error
+	rs, err = cc.ctx.ExecuteStmt(ctx, stmt)
+	cs, ok := cc.ctx.GetSessionVars().GetCache(stmt)
```

Comment:
Why getting the cache after `ExecuteStmt`?

Reply:
In order to avoid the following situations to choose the wrong result：
use test1
select * from t1
use test2
select * from t1

Reply:
Sorry, I don't get this...if the statement has been executed, why do we need to get a cache?

Reply:
The actual execution of the query is done in rs.next(). like
<img width="351" alt="image" src="https://user-images.githubusercontent.com/97376271/212102000-300b65ad-a23c-4e62-80ae-6dbd214dfde4.png">

Reply:
I got your point, thanks!